src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/FormattingContext.cs

```diff
@@ -19,20 +22,20 @@ namespace Microsoft.AspNetCore.Razor.LanguageServer.Formatting
 {
     internal class FormattingContext : IDisposable
     {
-        private AdhocWorkspaceFactory _workspaceFactory;
-        private Document _csharpWorkspaceDocument;
-        private AdhocWorkspace _csharpWorkspace;
+        private readonly AdhocWorkspaceFactory _workspaceFactory;
+        private Document? _csharpWorkspaceDocument;
+        private AdhocWorkspace? _csharpWorkspace;
 
         private FormattingContext(AdhocWorkspaceFactory workspaceFactory)
         {
             _workspaceFactory = workspaceFactory;
         }
 
-        public DocumentUri Uri { get; private set; }
+        public DocumentUri Uri { get; private set; } = null!;
```

Comment:
I verified these properties are all initialized in the `Create` method, so this is safe. Required properties in C# 10 should mean we can remove the initialization.